driver/pparse.ml

```diff
@@ -79,9 +85,10 @@ let apply_rewriter magic fn_in ppx =
   end;
   fn_out
 
-let read_ast magic fn =
+let read_ast (type a) (kind : a ast_kind) fn : a =
   let ic = open_in_bin fn in
   try
+    let magic = magic_of_kind kind in
     let buffer = really_input_string ic (String.length magic) in
     assert(buffer = magic); (* already checked by apply_rewriter *)
```

Comment:
It's not specific to this PR, but since `read_ast` is exposed in the mli and can be used in other contexts, it should raise a proper error rather than an assertion failure